pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -170,6 +199,20 @@ func corefileChanged(current, expected *corev1.ConfigMap) (bool, *corev1.ConfigM
 	return true, updated
 }
 
+func coreDNSResolver(upstream operatorv1.Upstream) (string, error) {
+	if upstream.Type == operatorv1.NetworkResolverType {
+		if upstream.Address == "" {
+			return "", errInvalidNetworkUpstream
+		}
+		if upstream.Port > 0 {
+			return fmt.Sprintf("%s:%d", upstream.Address, upstream.Port), nil
```

Comment:
The address could be IPv6, right?  
```suggestion
			return net.JoinHostPort(upstream.Address, fmt.Sprintf("%d", upstream.Port)), nil
```

Reply:
great catch! thx